scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/AvroUtils.java

```diff
@@ -37,31 +38,28 @@
 class AvroUtils {
   private AvroUtils() {}
 
-  public static String[] toKeyPath(String keyField) {
-    return keyField.split("\\.");
-  }
-
-  public static String validateKeyField(String keyField, Class<?> keyClass, Schema schema) {
-    final String[] keyPath = toKeyPath(keyField);
+  public static int[] toKeyPath(String keyField, Class<?> keyClass, Schema schema) {
```

Comment:
Can you add a docstring about how the `int[]` should be interpreted? Assuming it's a series of avro field indexes...